### DISCUSSION TITLE
Typical power usage for SpanPoint

### DISCUSSION THREAD

**Discussion by YouCanNotBeSerious** [HomeSpan/HomeSpan#700] [Q&A] [answered]
I have been reading and can't (yet?) find an answer - basically, is SpanPoint able to realistically run battery powered remote sensors with reasonable battery life? Anyone got real world examples?

My use case would be door/window sensors, so I want timely notification i.e. no more than a couple of seconds delay, something else I am trying to answer is if polling is needed (as shown in the example) or an interrupt based wakeup can be done. Speaking of wakeup - it looks like all power management is to be added by me from the example I see (temperature sensor example) is that correct?   That's fine, but just checking I'm not about to badly duplicate something already existing!

**Comment by HomeSpan** [comment, marked as answer]
Polling should be done on the main device, which is plugged in so there is no need to conserve power.  But the remote sensor is battery operated and needs to conserve as much as possible.  In my example I show how you can stay in sleep mode and wake every 30 seconds to quickly take temperature, transmit, and drop back into sleep mode.  But if you are checking a window/door sensor, waking up every 30 seconds is probably too long.  May need to shorten that to 5 seconds, but that will reduce battery life.  In my own tests, I only use a simple rechargeable battery, not a proper alarm/sensor battery, which could last a lot longer.  Alternatively, you can program the remote sensor to wake on a pin becoming active (via a magnetic switch).  This would enable you to stay in sleep mode until an event occurs.  I've not explored this method.

Short answer is to experiment - won't know till you try :).

**Comment by YouCanNotBeSerious** [comment]
Thanks - I will finish up my initial familiarity and testing and do some experiments!

**Comment by homonto** [comment]
just to add to what @HomeSpan  said: I have built some devices a year ago and they survived almost a year on 500mAh battery.
Some of them are "wake up every 5min and send data to bridge" but others are "motion detector ONLY" (that can be also: door/window open) so not waking up regularly - only when event occurs.  These ones survive depending where the sensor is: the kitchen one dies in months but another one, in the room that is rarely used, more than 12 months and still working. But this also depends on your sensor design i.e.:
- which LDO (some LDO consume 25uA, some 5uA)
- which PIR sensor - this is even worse: some consume 400uA. But there are also with 3uA
- if you use some tricks: using i.e. "TPL5111 Nano-Power SystemTimer for Power Gating" or TPL5110 - then you can completely power off the ESP32 and wake up it on trigger to TPL - then you can be really successful with battery life
- for window/door open sensor the PIR issue is not an issue of course

see below the battery life in one of them (every 5min wake up):
<img width="1533" alt="Screenshot 2023-12-23 at 07 38 17" src="https://github.com/HomeSpan/HomeSpan/assets/46562447/0d167abc-194e-4370-bbd7-aa4a0f15bc2d">

**Reply by YouCanNotBeSerious** [reply]
Great info, thanks for taking the time to reply.  
I have gotten to the stage of coding an interrupt wake from sleep slave device which I am just finishing up designing some 3D printed cases for and then it will be a door sensor. I will have it in an area where it should wake at least a dozen times per day, and will be tracking battery life.  The board is a Lolin C3 pico with I believe all the power saving hardware in place,   _I don't know how to ID the regulator but the fact they have pads that you need to cut/join to enable LEDs and things makes me think they took care in part selection :-)_  FOUND the regulator is a ME6211C33 with advertised "Standby Current: 0.1uA"

**Reply by homonto** [reply]
ha, so majority of my remotes are on C3 (ESP32 is too slow, ESP32-S2 is good, ESP32-S3 is overkill)

but to reduce completely the power consumption I don't use dev boards - I use the small module:
<img width="326" alt="Screenshot 2023-12-23 at 07 47 40" src="https://github.com/HomeSpan/HomeSpan/assets/46562447/7403fb5e-fee7-4138-bef7-fcd61fed351c">

and this is how it looks inside:
![20231209_122048](https://github.com/HomeSpan/HomeSpan/assets/46562447/eeff3a12-3155-4b35-b9b6-406f75af35ed)
LDO I am using is: XC6220B331 3.3V - only 6uA on standby 

and for light: TSL2561 and for temp/hum: SHt31 - they are both not powered while sleeping -the got powered on wake up directly from one of GPIO

**Reply by homonto** [reply]
I tested ME6211 - very bad, around 30uA. But that might be just mine. So I moved to XC6220B331

**Reply by YouCanNotBeSerious** [reply]
Actually the dat sheet does say " Low Power Consumption: 40uA(TYP.)
 Standby Current: 0.1uA(TPY.)" so the 40uA matches your testing close enough - when I look closely at the data sheet, what the manufacturer means by standby is literally the regulator by itself in sleep/standby/shutdown mode, not supplying the 3V3 output which is not how it will be in real life for me.

I am going to go ahead with my tests using this module and see, but thank you for your data.

**Comment by homonto** [comment]
see the mentioned motion sensor data  (ESP32-S2) - las time charged: October 2022 and still 38% of battery. But again: it wakes up rarely.

<img width="336" alt="Screenshot 2023-12-23 at 08 02 13" src="https://github.com/HomeSpan/HomeSpan/assets/46562447/35d621bb-bc2c-4137-a67e-ff4a37d49962">
